packages/x-data-grid/src/hooks/core/useGridRefs.ts

```diff
@@ -2,10 +2,10 @@ import * as React from 'react';
 import type { GridPrivateApiCommon } from '../../models/api/gridApiCommon';
 
 export const useGridRefs = <PrivateApi extends GridPrivateApiCommon>(
-  apiRef: React.MutableRefObject<PrivateApi>,
+  apiRef: React.RefObject<PrivateApi>,
 ) => {
   const rootElementRef = React.useRef<HTMLDivElement>(null);
-  const mainElementRef = React.useRef<HTMLDivElement | null>(null);
+  const mainElementRef = React.useRef<HTMLDivElement>(null);
```

Comment:
`useRef` handles the `| null` typing when `null` is passed as the default value.